Restore dashboards: Add documentation (#112272)
grafana/grafana: docs/sources/dashboards/manage-dashboards/index.md

The new folder name is automatically saved.

## Dashboard and folder deletion

When you delete folders, the action is immediate and can't be reversed.

Deleted dashboards, however, are stored in the deletion history and can be [restored](#restore-dashboards) if needed.

## Restore dashboards

{{% admonition type="caution" %}}
The feature is only available in Grafana Cloud.  
{{% /admonition %}}

> When you delete dashboards, they’re stored in the deletion history for **up to 12 months**, allowing you to restore them if needed.  

Reviewer: ```suggestion
When you delete dashboards, they’re stored in the deletion history for up to 12 months, allowing you to restore them if needed.
```

> The deletion history can contain a maximum of **1000 dashboards** — once this limit is reached, the oldest deleted dashboards may be permanently removed before the 12-month period expires.

Reviewer: ```suggestion
The deletion history can contain a maximum of 1000 dashboards.
Once this limit is reached, the oldest deleted dashboards are permanently removed even if the 12-month expiration period isn't over yet.
```
